File: from-1301-to-1400/1325-delete-leaves-with-a-given-value/solution.py
# Definition for a binary tree node.
# class TreeNode:
#     def __init__(self, val=0, left=None, right=None):
#         self.val = val
#         self.left = left
#         self.right = right
class Solution:
    def removeLeafNodes(self, root: Optional[TreeNode], target: int) -> Optional[TreeNode]:
        def dfs(node: TreeNode, tgt: int)-> bool:
            if not node:
                return True
    
            if dfs(node.left, tgt):
                node.left = None
            if dfs(node.right, tgt):
                node.right = None
            if node.val == tgt and not node.left and not node.right:
                return True

            return False

        dfs(root, target)
        if not root.left and not root.right and root.val == target:
            return None

        return root

File: from-1301-to-1400/1325-delete-leaves-with-a-given-value/solution.go
/**
 * Definition for a binary tree node.
 * type TreeNode struct {
 *     Val int
 *     Left *TreeNode
 *     Right *TreeNode
 * }
 */
 func removeLeafNodes(root *TreeNode, target int) *TreeNode {
    var dfs func(node *TreeNode, tgt int) bool 
    dfs = func(node *TreeNode, tgt int) bool {
        if node == nil {
            return true
        }
        if dfs(node.Left, tgt) {
            node.Left = nil
        }
        if dfs(node.Right, tgt) {
            node.Right = nil
        }
        if node.Val == tgt && node.Left == nil && node.Right == nil {
            return true
        }
        return false
    }
    dfs(root, target)
    if root.Left == nil && root.Right == nil && root.Val == target {
        return nil
    }
    return root
}